packages/core/src/browser/context-menu-renderer.ts

```diff
@@ -94,15 +94,14 @@ export interface RenderContextMenuOptions {
 export namespace RenderContextMenuOptions {
     export function resolve(menuPathOrOptions: MenuPath | RenderContextMenuOptions, anchor?: Anchor, onHide?: () => void): RenderContextMenuOptions {
         let menuPath: MenuPath;
-        let args: any[];
+        let args: any[] | undefined;
         if (Array.isArray(menuPathOrOptions)) {
             menuPath = menuPathOrOptions;
-            args = [anchor!];
         } else {
             menuPath = menuPathOrOptions.menuPath;
             anchor = menuPathOrOptions.anchor;
             onHide = menuPathOrOptions.onHide;
-            args = menuPathOrOptions.args ? [...menuPathOrOptions.args, anchor] : [anchor];
+            args = menuPathOrOptions.args;
```

Comment:
I think this change is correct, but I happen to know that I've relied on the existing behavior downstream to distinguish command activations from different sources, and I may not be alone in that. A more limited fix to the specific problem might be to check whether the type of the arg is string before setting it in the search bar of the preferences view.

Reply:
That was my first thought as well, but that sounds like a workaround, as this issue will repeat itself for any command with a single optional argument. But I guess I can change it that way, if you rely on it. Maybe there's a cleaner solution for that problem?

Reply:
I'm OK with removing it - I know where I've relied on it and how to fix it - but it would definitely be a breaking change worth highlighting in the changelog, and I'm not sure that there aren't legitimate uses for it that we'd be cutting off. It's worth being cautious about, because it isn't something that people may realize was questionable behavior, and they won't get much help from TypeScript about it, because command handlers are supposed to expect `any[]`, so changes in what will _actually_ be passed in aren't detected by the type system.